### ISSUE TITLE
Solar panel to buck regulator to linear regulator?

### ISSUE THREAD

**Issue by romankc** [vitzaoral/esp32_beehive#1]
I see that you are wanting to charge a Lithium Ion battery from a solar panel.
In your current design, you will experience some efficiency loss in the DC-DC converter and additional loss in the linear charge controller, about 15% total loss in the very best case scenario and 30% typically.

Why not use a DC-DC step down battery charging IC instead?

**Comment by vitzaoral** [comment]
Hello, thanks for your comment.
I am not sure if I correctly understand which part of design is wrong.

Battery charging is done by the [TP4056 charging board](https://www.aliexpress.com/item/Smart-Electronics-5V-Micro-USB-1A-18650-Lithium-Battery-Charging-Board-With-Protection-Charger-Module-for/32500042672.html). According to [this video](https://www.youtube.com/watch?v=dBx-g1dkdDQ) is most efficient. Unfortunately, the maximum input for TP4056 is 6V, then it does not work. Solar panels on direct sun can get up to 7V, so I used this[ DC-DC Step Down](https://www.aliexpress.com/item/Ultra-Small-Size-DC-DC-Step-Down-Power-Supply-Module-3A-Buck-Converter-Adjustable-1-8V/32880983608.html) regulator to convert solar output to 5V. There is some efficiency loss.

Then I need to power ESP32 with 3.3V. Voltage of battery is between 4.2V - 3.0V. I tried several DC-DC step down regulator and finally I used [DD0503MA](https://www.aliexpress.com/item/Ultra-Small-mini-DC-3-7V-4-2V-4-5V-5V-6V-to-3-3V-Step/32654976787.html) 3.3V regulator to power ESP32 with battery.  There is some efficiency loss too.

So there are two regulators - one for the TP4056 and other one for the ESP32. If you have some advice how to design power managing better, I will be grateful.
Thank you.

**Comment by romankc** [comment]
For powering the ESP32, consider these: TPS62740, TPS78330, TPS62203.
You can achieve higher efficiency by using an IC which is specifically designed for low levels of current.
At these low current levels, and especially while the device is sleeping, the quiescent current is very important.

I know of that YouTube video. Unfortunately the entire test process that he used is wrong so the results are wrong.

Suppose your solar panel outputs 7V and your battery has a voltage of 3.7V before you begin charging.
We'll suppose that your DC-DC converter has negligible loses.
You feed the TP4056 with 5V and it regulates down to 3.7V.
3.7/5=74%.

You lose at least 26% of your solar panel power!

Even if the battery is almost fully charged, 4.2V/5V, you still lose at least 16% on the TP4056 at an input of 5V.

This is why you need to use a step down converter to go directly from the solar panel to the battery.

What is the current rating of your panel?

**Comment by vitzaoral** [comment]
Thank you for your reply, I will consider using the TPSXXXXX converters, I did not know about them.

I am a 90% software developer, so I do not have enough knowledge of how to design a solid circuit (at least it works:) But I understand what you're saying. So you propose to remove TP4056 and use convertor directly to charging battery.

I am using [this solar panels](https://www.aliexpress.com/item/BUHESHUI-6V-1000mA-6W-Mini-Monocrystalline-PET-Solar-Panel-Small-Solar-Cell-Battery-Bicycle-Sharing-Share/32832618716.html) - 6V 1000mA 6W. I am using two of them (parallel connected). Battery is [Li-ion 18650 3400mAh](https://www.aliexpress.com/item/2-PCS-lot-New-Protected-Original-Rechargeable-battery-18650-NCR18650B-3400mah-with-PCB-3-7V-Free/32628168837.html).

**Comment by romankc** [comment]
[This](https://www.aliexpress.com/item/1S-Cell-Li-ion-Lithium-Battery-lipo-battery-Charger-Board-Module-3-7V-4-2V-CN3791/32862470804.html), or any other CN3791 based module, may work well for you. Just make sure that you get the 6V version. The 9V and 12V versions appear to be half the price and you can convert them to 6V by changing one resistor.

Compare it side by side with your design and let me know the results.

Hint: Stick an INA219 before and after the charge controller to measure the efficiency.

**Comment by romankc** [comment]
Oh, also try your panels in series (with a 12V CN3791) and see if this gives you better or worse overall power production.

**Comment by vitzaoral** [comment]
OK, so I ordered CN3791 (6V and 12V version) and I will try it with my solar panels in parallel and series connection. I assume that the Zener diode is no longer needed behind the solar panel (for backflow protection), which also has some loss of efficiency. I wonder if the 6V version will work when the panels in full sun give around 7V. In the comment on AliExpress someone wrote that for his 12V version it is turning off when the input voltage is over 12.5V..

Delivery from Aliexpress takes about 2 weeks to one month, so now I have to wait, at least I can form the software part of the project...

Thank you very much for your advices, I appreciate it!

**Comment by vitzaoral** [comment]
Hi @romankc .
So finally I use 2x 6V solar panels connected in series with 12V version CN3791 and it works great. I think it works better than 6V version. Thank you.
